File: src_OLD/ShortStackServer/JsonRpcTypes/OriginBranchInformation.cs
// <copyright file="OriginBranchInformation.cs" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

namespace ShortStackServer.JsonRpcTypes
{
    using System.Runtime.Serialization;

    /// <summary>
    /// Encapsulates information about a remote GIT branch.
    /// </summary>
    [DataContract]
    public class OriginBranchInformation
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="OriginBranchInformation"/> class.
        /// </summary>
        /// <param name="friendlyName">The friendly name of the branch.</param>
        /// <param name="remoteName">The remote name of the branch.</param>
        /// <param name="canonicalName">The canonical name of the branch.</param>
        public OriginBranchInformation(string friendlyName, string remoteName, string canonicalName)
        {
            this.FriendlyName = friendlyName;
            this.RemoteName = remoteName;
            this.CanonicalName = canonicalName;
        }

        /// <summary>
        /// Gets the friendly name of the branch.
        /// </summary>
        [DataMember(Name = "friendlyName")]
        public string FriendlyName { get;  }

        /// <summary>
        /// Gets the remote name of the branch.
        /// </summary>
        [DataMember(Name = "remoteName")]
        public string RemoteName { get; }

        /// <summary>
        /// Gets the canonical name of the branch.
        /// </summary>
        [DataMember(Name = "CanonicalName")]
        public string CanonicalName { get; }
    }
}

File: src_OLD/VSCodeExtension/src/JsonRpcTypes/OriginBranchInformation.ts
/* --------------------------------------------------------------------------------------------
 * Copyright (c) Microsoft Corporation. All Rights Reserved.
 * ------------------------------------------------------------------------------------------ */

 /**
 * Used by the JSON-RPC to say hello world to the caller.
 */
namespace JsonRpcTypes {
    export interface OriginBranchInformation {
        /**
         * Gets the friendly name of the branch.
         */
        friendlyName : string;

        /**
         * Gets the remote name of the branch.
         */
        remoteName : string;

        /**
         * Gets the canonical name of the branch.
         */
        canonicalName : string;
    }
}